### PR TITLE
Override std out for application

### PR DESCRIPTION
**Pull request by KIRA009**
**What kind of change does this PR introduce?**
This PR addresses #804 

**Summary**
This PR wraps the entire app in a context processor, that redirects all stdout and stderr to a log file

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have performed a self-review of my code
* [ ] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**
Build the app, then try and run it. You should be seeing 

**Other information**

### CODE REVIEW

File: openadapt/build_utils.py [hunk 1]
```diff
@@ -36,10 +36,11 @@ class RedirectOutput:
     def __enter__(self) -> "RedirectOutput":
         """Redirect stdout and stderr to /dev/null."""
         if is_running_from_executable():
-            null_stream = open(os.devnull, "a")
+            log_file_path = get_root_dir_path() / "data" / "openadapt.log"
```
Comment (abrichr) [inline openadapt/build_utils.py:39]: What do you think about adding the date/time to the filename?

File: openadapt/entrypoint.py [hunk 2]
```diff
@@ -13,21 +13,19 @@
...
+            from openadapt.config import print_config
 
-        print_config()
-        load_alembic_context()
-        tray._run()
-    except Exception as exc:
-        data_dir = get_root_dir_path()
-        # TODO: log all exceptions to a file
-        with open(data_dir / "error.log", "a") as f:
-            f.write(f"{datetime.now()}: {exc}\n")
```
Comment (abrichr) [inline openadapt/entrypoint.py:28]: Why was this removed? Why not log to the same file? e.g.

```
with redirect_stdout_stderr() as redirect:
    try:
        ...
    except:
        with open(redirect.log_file, "a") as f:
            ...
```

**Review by abrichr — COMMENTED** [review]
Thank you @KIRA009 ! Just a couple of small comments

File: openadapt/entrypoint.py (hunk 2 above)
Comment (KIRA009) [reply openadapt/entrypoint.py:28]: Since we are overriding stdout, print will automatically write to that file

File: openadapt/build_utils.py (hunk 1 above)
Comment (KIRA009) [reply openadapt/build_utils.py:39]: Makes sense. I was also wondering if we shpuld separate log files by versions?

Comment (abrichr) [reply openadapt/build_utils.py:39]: I think as long as the version is logged, date/time in the path is sufficient.

File: openadapt/entrypoint.py (hunk 2 above)
Comment (abrichr) [reply openadapt/entrypoint.py:28]: Why `print` and not `logger.info`?

File: openadapt/entrypoint.py [hunk 3]
```diff
@@ -13,21 +13,19 @@
...
+            print_config()
+            load_alembic_context()
+            tray._run()
+        except Exception as exc:
+            # TODO: log all exceptions to a file
+            print(f"{datetime.now()}: {exc}")
```
Comment (abrichr) [inline openadapt/entrypoint.py:27]: Why `print` and not `logger.exception(exc)`?

File: openadapt/build_utils.py (hunk 1 above)
Comment (KIRA009) [reply openadapt/build_utils.py:39]: Should be easy, and also should help when in cases where the log file gets too large

**Review by abrichr — APPROVED** [review]
LGTM!
